lib/log.c

```diff
@@ -868,11 +868,16 @@ int zlog_rotate(void)
 		save_errno = errno;
 		umask(oldumask);
 		if (zl->fp == NULL) {
+
```

Comment:
I think you had already merged these changes. These changes looks good, but instead of unlocking and locking temporarily like below,  would it be useful to if

1) use pthread_mutex_trylock inside zlog_rotate or vzlog instead of pthread_mutex_lock
or
2) if we convert this mutex as a recursive mutext

Thanks.

Reply:
These changes are in fact already merged. In any case, neither of those changes are unnecessary; I think this is the correct way to fix the problem...